src/pages/MyPage/Edit/Edit.js

```diff
@@ -0,0 +1,217 @@
+
+  console.log(data);
+  const isValid = useMemo(() => {
+    const validations = {};
+    if (data) {
+      for (let key of Object.keys(data)) {
+        validations[key] = true;
+      }
+    }
+    return validations;
+  }, [data]);
+
+  const typing = (e) => {
+    const { name, value } = e.target;
+    setData({ ...data, [name]: value });
+  };
+
+  const setGender = (e) => {
+    setData({ ...data, gender: e.target.value });
+  };
+
+  const goInfo = () => {
+    navigate('/');
+  };
+  const closePopup = () => {
+    setPopup({ ...popup, open: false });
+  };
+
+  const editDelete = () => {
+    setPopup({
+      open: true,
+      title: '변경된 내용을 삭제하시겠어요.',
+      leftBtnValue: '네',
+      rightBtnValue: '아니오',
+      leftBtnClick: goInfo,
+      rightBtnClick: closePopup,
+    });
+  };
+
+  useEffect(() => {
+    fetch('http://10.58.52.69:8000/users', {
+      headers: {
+        'Content-Type': 'application/json;charset=utf-8',
+        Authorization: token,
+      },
+    })
+      .then((res) => {
+        return res.json();
+      })
+      .then((result) => {
+        setData(result.data);
+      });
+  }, []);
+
+  const InfoEdit = () => {
+    const values = Object.values(data);
+    for (let i = 0; i < values.length; i++) {
+      if (values[i] === '') {
+        setPopup({
+          open: true,
+          title: '기입하지 않은 값이 있네요.',
+          leftBtnValue: '계속 작성하기',
+          rightBtnValue: '수정 취소하기',
+          leftBtnClick: closePopup,
+          rightBtnClick: goInfo,
+        });
+        return;
+      }
+    }
+
+    const validValues = Object.values(isValid);
+    for (let j = 0; j < validValues.length; j++) {
+      if (!validValues[j]) {
+        setPopup({
+          open: true,
+          title: '유효하지 않은 값이 있네요.',
+          leftBtnValue: '계속 작성하기',
+          rightBtnValue: '수정 취소하기',
+          leftBtnClick: closePopup,
+          rightBtnClick: goInfo,
+        });
+        return;
+      }
+    }
+
+    setPopup({
+      open: true,
+      title: '변경된 내용을 적용하시겠어요.',
+      leftBtnValue: '적용하기',
+      rightBtnValue: '취소하기',
+      leftBtnClick: userInfoSubmit,
+      rightBtnClick: closePopup,
+    });
+  };
+
+  const userInfoSubmit = () => {
+    fetch('http://10.58.52.69:8000/users', {
+      method: 'PUT',
+      headers: {
+        'Content-Type': 'application/json;charset=utf-8',
+        Authorization: token,
+      },
+      body: JSON.stringify({
+        badgeImageUrl: data.badgeImageUrl,
+        userProfile: data.userProfile,
+        nickname: data.nickname,
+        age: data.age,
+        gender: data.gender,
+        height: data.height,
+        weight: data.weight,
+        skeletalMuscleMass: data.skeletalMuscleMass,
+        bodyFat: data.bodyFat,
+        goalWeight: data.goalWeight,
+        goalBodyFat: data.goalBodyFat,
+        goalSkeletalMuscleMass: data.goalSkeletalMuscleMass,
+        badgeLevel: data.badgeLevel,
+        isSubscribe: data.isSubscribe,
+      }),
```

Comment:
data state의 key값이 보내야 할 값들과 완전히 동일하다면, 그냥 data 객체 자체를 보내시면 됩니다.
```suggestion
      body: JSON.stringify(data),
```

Reply:
수정 완료요